internals-js/src/operations.ts

```diff
@@ -1001,16 +1019,17 @@ export class Operation extends DirectiveTargetElement<Operation> {
       }
     }
 
-    return this.withUpdatedSelectionSetAndFragments(optimizedSelection, finalFragments ?? undefined);
+    return this.withUpdatedSelectionSetAndFragments(optimizedSelection, finalFragments ?? undefined, allAvailableVariables);
   }
 
-  generateQueryFragments(): Operation {
+  generateQueryFragments(allAvailableVariables: VariableDefinitions = new VariableDefinitions()): Operation {
```

Comment:
I'm confused how fragment generation for an operation could introduce fragments with unseen variables (there's a case to be made for fragment reuse, where they can introduce fields that are never executed at runtime but introduce references, but we can't say the same for generation).

Reply:
I think you may be right about that. I had been thinking that we need to cover our bases by doing it for both paths, but I think this isn't necessary. I'll take it out.